### PR TITLE
Use cudnn maxpool when possible

### PR DESCRIPTION
**Pull request by Flowingsun007**

### CODE REVIEW

File: oneflow/user/kernels/pooling_kernel.cu [hunk 1]
```diff
@@ -22,7 +22,7 @@ limitations under the License.
 namespace oneflow {
 namespace {
 
-constexpr int kBlockSize = cuda::elementwise::kBlockSize << 2;
+constexpr int kBlockSize = cuda::elementwise::kBlockSize;
```
Comment (strint) [inline oneflow/user/kernels/pooling_kernel.cu:25]: kBlockSize 为什么改小呢

**Review by strint — APPROVED** [review]

**Comment by strint** [comment]
https://github.com/Oneflow-Inc/models/tree/main/Vision/classification/image/resnet50

```
python3 -m oneflow.distributed.launch --nproc_per_node 1 --nnodes 1 --node_rank 0 --master_addr 127.0.0.1 Vision/classification/image/resnet50/train.py --ofrecord-path $OFRECORD_PATH --ofrecord-part-num 1 --num-devices-per-node 1 --lr 0.004 --momentum 0.875 --num-epochs 1 --train-batch-size 4 --val-batch-size 50 --print-interval 10 --exit-num 1 --use-fp16 --channel-last --scale-grad --graph --fuse-bn-relu --fuse-bn-add-relu --use-gpu-decode

```
参考这个启动脚本，做下验证吧

File: oneflow/user/kernels/pooling_kernel.cu (hunk 1 above)
Comment (Flowingsun007) [reply oneflow/user/kernels/pooling_kernel.cu:25]: 这个改动是为了恢复之前的设置

**Comment by ouyangyu** [comment]
> https://github.com/Oneflow-Inc/models/tree/main/Vision/classification/image/resnet50
> 
> ```
> python3 -m oneflow.distributed.launch --nproc_per_node 1 --nnodes 1 --node_rank 0 --master_addr 127.0.0.1 Vision/classification/image/resnet50/train.py --ofrecord-path $OFRECORD_PATH --ofrecord-part-num 1 --num-devices-per-node 1 --lr 0.004 --momentum 0.875 --num-epochs 1 --train-batch-size 4 --val-batch-size 50 --print-interval 10 --exit-num 1 --use-fp16 --channel-last --scale-grad --graph --fuse-bn-relu --fuse-bn-add-relu --use-gpu-decode
> ```
> 
> 参考这个启动脚本，做下验证吧

晚点等机器空闲了，再跑下速度。

**Comment by jackalcooper** [comment]
这个可以合并了吗

**Comment by ouyangyu** [comment]
> 这个可以合并了吗

可以。稍等一下贴结果。

**Comment by ouyangyu** [comment]
- ###  20220124 ResNet50 基于[7975e25](https://github.com/Oneflow-Inc/oneflow/commit/7975e25) 在V100上的性能测试结果

| Check            | Case Name  | Graph/Lazy                                                   |
| ---------------- | ---------- | ------------------------------------------------------------ |
|                  | gpudecode_FP16_b512_1n1g         |                                                              |
| Y | Throughput | [1410.46](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCgpu_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_085852074646314.log) / [1402.226](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_091524725959161.log) ≈ 100.587 % |
| Y     | Memory     | [23640](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCgpu_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_085852074646314.log) MiB / [23892](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_091524725959161.log) MiB ≈ 98.945 %  |
|                  | cpudecode_FP16_b512_1n1g         |                                                              |
| N | Throughput | [1378.81](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCcpu_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_090453640307425.log) / [1433.386](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_092145509001247.log) ≈ 96.193 % |
| Y     | Memory     | [22878](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCcpu_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_090453640307425.log) MiB / [23128](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP16_b512_1n1g_use_cudnn_maxpool_when_possible_20220124_092145509001247.log) MiB ≈ 98.919 %  |
|                  | gpudecode_FP32_b256_1n1g         |                                                              |
| Y | Throughput | [378.86](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCgpu_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_090117481903812.log) / [381.494](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_091801473401506.log) ≈ 99.31 % |
| Y     | Memory     | [23718](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCgpu_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_090117481903812.log) MiB / [23450](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_091801473401506.log) MiB ≈ 101.143 %  |
|                  | cpudecode_FP32_b256_1n1g         |                                                              |
| Y | Throughput | [380.83](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCcpu_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_090722185823177.log) / [383.634](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_092421379889586.log) ≈ 99.269 % |
| Y     | Memory     | [22972](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-graph/dlperf/1n1g/resnet50_graph_DCcpu_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_090722185823177.log) MiB / [22764](https://oneflow-test.oss-cn-beijing.aliyuncs.com/OneBrain/commit/use_cudnn_maxpool_when_possible/20220124/7975e25/ResNet50-lazy/dlperf/1n1g/resnet50_lazy_FP32_b256_1n1g_use_cudnn_maxpool_when_possible_20220124_092421379889586.log) MiB ≈ 100.914 %  |

**Comment by strint** [comment]
> 20220124 ResNet50 基于7975e25 在V100上的性能测试结果

已经恢复了之前最好的结果。

**Comment by Flowingsun007** [comment]
> > 20220124 ResNet50 基于7975e25 在V100上的性能测试结果
> 
> 已经恢复了之前最好的结果。

好的，麻烦ouyang点一下apporve吧

**Pull request merged**